File: 프로그래머스/1/12982. 예산/예산.py
def solution(d, budget):
    answer = 0
    
    # 먼저 부서를 오름차순으로 정렬
    d.sort()
    
    # 정렬된 부서를 순환하면서 
    for i in d:
        if budget >= i: # 만약 예산이 i보다 크면
            budget -= i  # 예산에서 i만큼 차감하고 answer 에 1 카운트
            answer += 1
        else: # 예산이 충분하지 않다면 끝
            break

    return answer

File: 프로그래머스/1/12982. 예산/예산.js
function solution(d, budget) {
    var answer = 0;
    
    // d 배열을 오름차순으로 정렬
    d.sort(function(a, b) {return a-b})
    
    // 오름차순 배열을 순회하면서 
    for (i = 0; i < d.length; i++) {
        // 변수 d의 i번째 요청 예산이 budget보다 작아질때까지 빼준다
        if (budget >= d[i] ) {
            budget -= d[i];
            answer += 1;
        } else { // d[i] 보다 변수가 작아지면 반복분을 빠져나온다
            break;
        }   
    }
    return answer;
}